File: src/Backend/TimeReport.Api/Data/Entities/DailyNote.cs
namespace TimeReport.Api.Data.Entities;

public class DailyNote
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public string Date { get; set; } = "";
    public string Content { get; set; } = "";
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public User User { get; set; } = null!;
}

File: railsapp/app/models/daily_note.rb
class DailyNote < ApplicationRecord
  self.table_name = "daily_notes"

  belongs_to :user

  validates :date, presence: true, format: { with: /\A\d{4}-\d{2}-\d{2}\z/ }
  validates :content, presence: true
  validates :date, uniqueness: { scope: :user_id }
end
